packages/muelu/test/unit_tests/ZeroSubBlockAFactory.cpp

```diff
@@ -0,0 +1,126 @@
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact
+//                    Jonathan Hu       (jhu@sandia.gov)
+//                    Andrey Prokopenko (aprokop@sandia.gov)
+//                    Ray Tuminaro      (rstumin@sandia.gov)
+//
+// ***********************************************************************
+//
+// @HEADER
+#include <Teuchos_UnitTestHarness.hpp>
+#include <Teuchos_DefaultComm.hpp>
+
+#include "MueLu_TestHelpers.hpp"
+#include "MueLu_Version.hpp"
+
+#include <Xpetra_BlockedCrsMatrix.hpp>
+
+#include "MueLu_FactoryManager.hpp"
+#include "MueLu_Utilities.hpp"
+#include "MueLu_ZeroSubBlockAFactory.hpp"
+
+namespace MueLuTests {
+
+  TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL(ZeroSubBlockAFactory, ConstructZeroSubBlockAFactory, Scalar, LocalOrdinal, GlobalOrdinal, Node)
+  {
+#   include "MueLu_UseShortNames.hpp"
+    MUELU_TESTING_SET_OSTREAM;
+    MUELU_TESTING_LIMIT_SCOPE(Scalar,GlobalOrdinal,NO);
+
+    out << "version: " << MueLu::Version() << std::endl;
+
+    RCP<ZeroSubBlockAFactory> subBlockAFactory = rcp(new ZeroSubBlockAFactory());
+    TEST_ASSERT(!subBlockAFactory.is_null());
+  }
+
+  TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL(ZeroSubBlockAFactory, ExtractZeroMainDiagonalBlockFrom2x2SaddlePointSystem, Scalar, LocalOrdinal, GlobalOrdinal, Node)
+  {
+#   include "MueLu_UseShortNames.hpp"
+
+    using test_factory = TestHelpers::TestFactory<SC, LO, GO, NO>;
+
+    MUELU_TEST_ONLY_FOR(Xpetra::UseTpetra)
```

Comment:
@trilinos/muelu This test should work for Epetra with `LocalOrdinal = int` and `GlobalOrdinal = int`, but I couldn't figure out how to limit the scope to this scenario. Other ordinal types seem not to implement the `CrsMatrix::expertStaticFillComplete()`. However, I couldn't figure out, how to limit the Epetra-version of this test to this particular scenario. I'd appreciate any help or hints. Thanks!

If somebody can assist with limiting the scope of this test as outlined above, I'd